src/utils/order_validation_utils.ts

```diff
@@ -12,6 +13,26 @@ export class OrderValidationUtils {
         this.tokenWrapper = tokenWrapper;
         this.exchangeWrapper = exchangeWrapper;
     }
+    public async validateOrderFillableThrowIfNotFillableAsync(
+        signedOrder: SignedOrder, zrxTokenAddress: string, expectedFillTakerTokenAmount?: BigNumber.BigNumber,
+    ): Promise<void> {
+        const orderHash = utils.getOrderHashHex(signedOrder);
+        const unavailableTakerTokenAmount = await this.exchangeWrapper.getUnavailableTakerAmountAsync(orderHash);
+        if (signedOrder.makerTokenAmount.eq(unavailableTakerTokenAmount)) {
```

Comment:
Can we make it that single error type is thrown only from one function?
And then this will look like:
```
this.validateIfExpired()
this.validateIfFillAmountZero();
```

I think - it will be more clear like that.